inspirehep/celery.py

```diff
@@ -25,15 +25,27 @@
 from __future__ import absolute_import, division, print_function
 
 import logging
+import os
 
-from flask_celeryext import create_celery_app
+from flask_celeryext import AppContextTask, create_celery_app
+from sqlalchemy.exc import InvalidRequestError, OperationalError
 
-from .factory import create_app
+from inspirehep.factory import create_app
 
 
-celery = create_celery_app(
-    create_app(LOGGING_SENTRY_CELERY=True)
-)
+LOGGER = logging.getLogger(__name__)
+
+
+class CeleryTask(AppContextTask):
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        if isinstance(exc, InvalidRequestError) or isinstance(exc, OperationalError):
+            LOGGER.error('Shutting down celery process because of {}', format(exc))
+            os.sys.exit(1)
```

Comment:
does this kill the worker process or the worker pool manager and hence the pod? the latter would be nicer as it gives more visibility and allows the pod to be rescheduled on a different node in case the node has issues.

Reply:
I tried to kill the PID 1 process to kill the container but apparently it's not possible.

Reply:
Thus, this kills only the celery process that is actually getting the task and running it.
As you can read from the steps I did for testing this, after the process is killed a new one is created from the pool.

Reply:
as figured out together, you can do
```suggestion
            os.kill(os.getppid(), signal.SIGTERM)
```

Reply:
cherry on the pie: set the exception message as terminattion reason https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#customizing-the-termination-message